de.tudarmstadt.ukp.wikipedia.timemachine/pom.xml

```diff
@@ -15,34 +15,46 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 -->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 	<parent>
-  <groupId>de.tudarmstadt.ukp.wikipedia</groupId>
-  <artifactId>de.tudarmstadt.ukp.wikipedia</artifactId>
-  <version>1.2.0-SNAPSHOT</version>
+		<groupId>de.tudarmstadt.ukp.wikipedia</groupId>
+		<artifactId>de.tudarmstadt.ukp.wikipedia</artifactId>
+		<version>1.2.0-SNAPSHOT</version>
 	</parent>
 	<modelVersion>4.0.0</modelVersion>
-  <artifactId>de.tudarmstadt.ukp.wikipedia.timemachine</artifactId>
-  <name>TimeMachine</name>
-  <dependencies>
-  	<dependency>
-  		<groupId>de.tudarmstadt.ukp.wikipedia</groupId>
-  		<artifactId>de.tudarmstadt.ukp.wikipedia.wikimachine</artifactId>
-  		</dependency>
-  	<dependency>
-  		<groupId>net.sf.trove4j</groupId>
-  		<artifactId>trove4j</artifactId>
-  	</dependency>
-  	<dependency>
-  		<groupId>de.tudarmstadt.ukp.wikipedia</groupId>
-  		<artifactId>de.tudarmstadt.ukp.wikipedia.mwdumper</artifactId>
-  	</dependency>
-  	<dependency>
-  		<groupId>log4j</groupId>
-  		<artifactId>log4j</artifactId>
-  	</dependency>
-  </dependencies>
-    	<build>
+	<artifactId>de.tudarmstadt.ukp.wikipedia.timemachine</artifactId>
+	<name>TimeMachine</name>
+	<dependencies>
+		<dependency>
+			<groupId>de.tudarmstadt.ukp.wikipedia</groupId>
+			<artifactId>de.tudarmstadt.ukp.wikipedia.wikimachine</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>net.sf.trove4j</groupId>
+			<artifactId>trove4j</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>de.tudarmstadt.ukp.wikipedia</groupId>
+			<artifactId>de.tudarmstadt.ukp.wikipedia.mwdumper</artifactId>
+		</dependency>
+		<!-- logger dependencies -->
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-log4j12</artifactId>
+			<scope>runtime</scope>
```

Comment:
These (`slf4j-log4j12`, `log4j `) should be test dependencies. The users of JWPL should decide themselves which logging backend they want to use being forced to add dependency excludes.

If these dependencies are required by some runnable JAR that might be created from this module, we should try finding some way that the dependency is included in the JAR but not transitively drawn in when the module is used as a library. E.g. by marking the dependencies as `provided` and configuring the fat-JAR generation such that `provided` dependencies are included.

Reply:
Update: They are required by the runnable JAR produced via the assembly-plugin. So we need to make sure, they are bundled into it, when set to `provided`. 

I'll have a look into the generated output and the assembly-plugin config, as - atm - they would not be bundled into the JAR file (with `provided`).

Reply:
Affected modules:

- datamachine
- timemachine
- revisionmachine

Reply:
@reckart Found a working solution that produces executable JARs which contain the required classes from the `provided` dependencies`.